tests/libcxxrt/libcxxrt.edl

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+enclave {
+   trusted {
+       public int test(
+           [in] const char** name);
```

Comment:
It seems like this ought to be an _out_ parameter.
Can you try returning a non zero value from test function in enc.cpp and see what happens?

Alternatively, I don't see why the test name needs to be returned from the enclave. We could just print the failing test name from the enclave itself.

Reply:
```suggestion
           [out] char** name);
```
I'm working on making this change.
I started off by adding an **else** clause in the **Test** function in **host.cpp** that does a **printf** of that value that returns for the **name** parameter.  **name** does indeed need to be an **out** parameter.
The tests all pass with the changes to libcxxrt.edl, host.cpp, and enc.cpp that only include changing name to out and removing const.
With the printf else clause in Test:

- test_foreign_exceptions passes
- test_guard fails
- test_typeinfo fails

printf produces the correct output in all cases.  I do not understand why this occurs.  I guess that there is some kind of cleanup and dangling read.

Reply:
I tried out the following changes off your branch and it seems to work fine:
https://github.com/Microsoft/openenclave/compare/ermumau04...anakrish/testport

What are you seeing?

Reply:
Yes.  It should work fine now.